src/editor/nodes/ParticleNode.js

```diff
@@ -0,0 +1,416 @@
+  };
+
+  static async deserialize(editor, json, loadAsync) {
+    const node = await super.deserialize(editor, json);
+
+    const {
+      src,
+      prewarm,
+      colorCurve,
+      velocityCurve,
+      startColor,
+      middleColor,
+      endColor,
+      startOpacity,
+      middleOpacity,
+      endOpacity,
+      emitterWidth,
+      emitterHeight,
+      size,
+      velocity,
+      endVelocity,
+      angularVelocity,
+      particleCount,
+      lifetime,
+      lifetimeRandomness
+    } = json.components.find(c => c.name === "particle").props;
+
+    node.startColor.set(startColor);
+    node.middleColor.set(middleColor);
+    node.endColor.set(endColor);
+    node.colorCurve = colorCurve;
+    node.velocityCurve = velocityCurve;
+    node.prewarm = prewarm;
+    node.startOpacity = startOpacity || 1;
+    node.middleOpacity = middleOpacity || 1;
+    node.endOpacity = endOpacity || 1;
+    node.emitterHeight = emitterHeight || 1;
+    node.emitterWidth = emitterWidth || 1;
+    node.lifetime = lifetime || 5;
+    node.size = size || 1;
+    node.lifetimeRandomness = lifetimeRandomness || 5;
+    node.particleCount = particleCount || 1000;
+    node.velocity.copy(velocity);
+    node.angularVelocity = angularVelocity || 0;
+    node.endVelocity = endVelocity || new THREE.Vector3(0, 0, 0.5);
+
+    loadAsync(
+      (async () => {
+        await node.load(src);
+      })()
+    );
+    node.createParticle();
+
+    return node;
+  }
+
+  static async load() {
+    defaultParticleSprite = await new Promise((resolve, reject) => {
+      new THREE.TextureLoader().load(defaultParticleUrl, resolve, null, e =>
+        reject(`Error loading Image. ${eventToMessage(e)}`)
+      );
+    });
+    defaultParticleSprite.flipY = false;
+  }
+
+  constructor(editor) {
+    const geometry = new THREE.BufferGeometry();
+    const material = new THREE.RawShaderMaterial({
+      uniforms: {
+        texture: { value: defaultParticleSprite }
+      },
+      vertexShader,
+      fragmentShader,
+      transparent: true,
+      depthWrite: false,
+      blendEquation: THREE.AdditiveBlending
+    });
+
+    super(editor, geometry, material);
+
+    this.lastUpdated = 0;
+    this._canonicalUrl = "";
+    this.emitterHeight = 1;
+    this.emitterWidth = 1;
+    this.initialPositions = [];
+    this.size = 1;
+    this.velocities = [];
+    this.velocity = new THREE.Vector3(0, 0, 0.5);
+    this.endVelocity = new THREE.Vector3(0, 0, 0.5);
+    this.angularVelocity = 0;
+    this.particleCount = 100;
+    this.lifetime = 5;
+    this.lifetimes = [];
+    this.lifetimeRandomness = 5;
+    this.ageRandomness = 5;
+    this.ages = [];
+    this.colors = [];
+    this.endColor = new THREE.Color();
+    this.middleColor = new THREE.Color();
+    this.startColor = new THREE.Color();
+    this.startOpacity = 1;
+    this.middleOpacity = 1;
+    this.endOpacity = 1;
+    this.prewarm = false;
+    this.colorCurve = 0;
```

Comment:
I think you need to set a valid initial value for `colorCurve` and `velocityCurve` so that the UI is properly set.